### PR TITLE
Add covariance/contravariance concept

### PR DESCRIPTION
**Pull request by dstockto**

### CODE REVIEW

**Comment by neenjaw** [comment]
I'll add the config files to the branch, this pr is awesome though

File: concepts/covariance-and-contravariance/about.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Introduction
+
+The concepts of covariance and contravariance in PHP are important as PHP enforces these rules are followed when extending classes or interfaces.
```
Comment (neenjaw) [inline concepts/covariance-and-contravariance/about.md:3]: Is there a small or limited example we can pull in to the introduction to illustrate this at a small scale?

**Review by neenjaw — COMMENTED** [review]
One question about the introduction, everything else is awesome

File: concepts/covariance-and-contravariance/about.md (hunk 1 above)
Comment (dstockto) [reply concepts/covariance-and-contravariance/about.md:3]: Would it be appropriate to put in maybe Calculator and InvalidCalc from the about.md? I'm not sure of a very brief way to introduce examples for this concept.

Comment (neenjaw) [reply concepts/covariance-and-contravariance/about.md:3]: Yea, maybe can add them in a single code block: a parent class, a child class that is valid under the rules and a child class that is invalid under the rules.

Also, did the contents of the `introduction` and `about` get reversed? Introduction should be more brief, about should be more detailed

Comment (dstockto) [reply concepts/covariance-and-contravariance/about.md:3]: It definitely looks like they got switched - I did have some weird merge conflicts but I thought I sanity checked too. I'll swap and get some examples into about.md

Comment (dstockto) [reply concepts/covariance-and-contravariance/about.md:3]: Er, into introduction.md

**Review by neenjaw — APPROVED** [review]

**Pull request merged**